src/metrics.jl

```diff
@@ -236,7 +236,7 @@ Base.@propagate_inbounds function _evaluate(d::UnionMetrics, a::AbstractArray, b
     end
     @inbounds begin
         s = eval_start(d, a, b)
-        if IndexStyle(a, b) === IndexLinear() || size(a) == size(b)
+        if IndexStyle(a, b) === IndexLinear() && eachindex(a) == eachindex(b)
```

Comment:
This is the crucial part: it makes sure that `eachindex(a, b)` in the next line is constructible.

Reply:
Looks like it makes no difference, but I guess you actually want to say `axes(a) == axes(b)`?

Reply:
You mean replacing the entire condition or just the second part?